api/envoy/api/v2/route/route.proto

```diff
@@ -553,24 +554,38 @@ message RouteAction {
   // backend).
   repeated HashPolicy hash_policy = 15;
 
-  // Indicates that a HTTP/1.1 client connection to this particular route
-  // should be allowed (and expected) to upgrade to a WebSocket connection. The
-  // default is false.
+  // Indicates that a HTTP/1.1 client connection to this particular route is allowed to
+  // upgrade to a WebSocket connection. The default is false.
   //
   // .. attention::
   //
-  //   If set to true, Envoy will expect the first request matching this route to
-  //   contain WebSocket upgrade headers. If the headers are not present, the
-  //   connection will be rejected. If set to true, Envoy will setup plain TCP
+  //   If a connection is upgraded to a WebSocket connection, Envoy will setup plain TCP
   //   proxying between the client and the upstream server. Hence, an upstream
   //   server that rejects the WebSocket upgrade request is also responsible for
   //   closing the associated connection. Until then, Envoy will continue to
   //   proxy data from the client to the upstream server.
   //
-  //   Redirects, timeouts and retries are not supported on routes where websocket upgrades are
-  //   allowed.
+  //   Redirects are not supported on routes where WebSocket upgrades are allowed.
   google.protobuf.BoolValue use_websocket = 16;
 
+  message WebSocketProxyConfig {
+    // See :ref:`stat_prefix <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.stat_prefix>`.
+    // If the parameter is not specified, the default value of "websocket" is used.
+    string stat_prefix = 1;
+
+    // See :ref:`idle_timeout <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.idle_timeout>`.
+    google.protobuf.Duration idle_timeout = 2
```

Comment:
you might want to add some text here, stating that the websocket connection will be terminated if inactive for X seconds and that this does not govern the inital upgrade stuff..

Reply:
Added.